src/plonk/circuit.rs

```diff
@@ -594,4 +606,70 @@ impl<F: Field> ConstraintSystem<F> {
         self.num_aux_columns += 1;
         tmp
     }
+
+    /// Hashes the `ConstraintSystem` into a `u64`.
+    pub fn hash(&self, mut hasher: &mut Blake2bState) -> [u8; 64] {
+        hasher.update(b"num_fixed_columns");
+        hasher.update(&self.num_fixed_columns.to_le_bytes());
+
+        hasher.update(b"num_advice_columns");
+        hasher.update(&self.num_advice_columns.to_le_bytes());
+
+        hasher.update(b"num_aux_columns");
+        hasher.update(&self.num_aux_columns.to_le_bytes());
+
+        hasher.update(b"num_gates");
+        hasher.update(&self.gates.len().to_le_bytes());
+        for gate in self.gates.iter() {
+            hasher.update(gate.0.to_owned().as_bytes());
```

Comment:
The name of the gate in code does not affect the argument, so this can be omitted.